packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/dataservice/DocumentController.java

```diff
@@ -128,6 +139,34 @@ public ResponseEntity<Integer> uploadDocumentFromGithub(
 
 	}
 
+	@PutMapping(value = "/{id}/uploadDocumentFromDOI")
+	public ResponseEntity<Integer> uploadDocumentFromDOI(
+		@PathVariable("id") String id,
+		@RequestParam("doi") String doi,
+		@RequestParam("filename") String filename
+	) {
+		try (CloseableHttpClient httpclient = HttpClients.custom()
+			.disableRedirectHandling()
+			.build()) {
+			byte[] fileAsBytes = downloadService.getPDF("https://unpaywall.org/" + doi);
```

Comment:
What is the success rate with using unpaywall? Should there be any sort of safety here in case they don't have a document?

Reply:
good point, I can make the doi and filename a required param

Reply:
as for the success rate - there are definitely some doi which we cannot be download (i couldn't tell you the rate) but the Document-Asset will have already been made on the front end without a pdf.  This should more or less mimic how these bad pdfs are being handled from the search

Reply:
Unpaywall doesn't return a PDF 100% of the time, so we need to handle a missing PDF case. With a notification to the user.

Reply:
added a notification is a pdf can't be uploaded